GSoC '24 - Dynamics Popup - Part 4

ketgg opened the pull request:
This is the fourth part of the main pull request: https://github.com/musescore/MuseScore/pull/23038, which adds the preview of dynamic on hovering.

PR for part 1: https://github.com/musescore/MuseScore/pull/24125
PR for part 2: https://github.com/musescore/MuseScore/pull/24147
PR for part 3: https://github.com/musescore/MuseScore/pull/24152

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

ketgg committed:
Add preview of dynamic on hover

ketgg committed:
Update itemRect to use QRectF

ketgg committed:
Add adjustedBoundingRect to set a fixed height for bounding rectangle

Jojo-Schmitz commented:
_GSoC '24 - Dynamics Popup - Part 4_ vs. _This is the third part of the main pull request_ ?!?

re1san commented:
> This is the third part of the main pull request ?!?

Oops! I forgot to edit, thanks for catching it @Jojo-Schmitz (_I was a bit sleepy yesterday_) 😅

cbjeukendrup committed:
VTests: use correct reference ref

The problem seems actually the opposite of what https://github.com/musescore/MuseScore/pull/24847/commits/1a8ded4d6c267e7064a3ca22725a16c9c229962c tried to solve
